### ISSUE TITLE
Provide a config source that accepts an existing config instance

### ISSUE THREAD

**Issue by tonybaines** [tonybaines/gestalt#22]
Something along the lines of 

```
Configurations
    .definedBy(TestConfig)
    .fromConfig(myConfigInstance)
    .fromXmlResource('common.xml')
    .done()
```

Where `myConfigInstance` would be some implementation of `TestConfig` from Gestalt or some other means.

Use cases would be
- Sourcing some configuration from another source e.g. a database or over HTTP
- Having a fallback for configuration data with extra behaviour (e.g. logging or other side-effects)

**Comment by tonybaines** [comment]
Implemented in [1.0.3](http://central.maven.org/maven2/com/github/tonybaines/gestalt/1.0.3)

**Comment by tonybaines** [comment]
implemented in 2.0.4
